src/processor/queue-consumer.service.ts

```diff
@@ -148,12 +148,13 @@ export class QueueConsumerService extends WorkerHost implements OnApplicationBoo
   private async checkCapacity(): Promise<void> {
     const capacityLimit = this.configService.getCapacityLimit();
     const capacity = await this.blockchainService.capacityInfo(this.configService.getProviderId());
+    const remainingCapacity = capacity.remainingCapacity;
     let outOfCapacity = false;
 
-    if (capacityLimit.type === 'percentage') {
+    if (capacityLimit.type === 'percentage' && remainingCapacity > 0) {
```

Comment:
`remainingCapacity` could be undefined in case a provider hasnt staked the capacity yet so this check will make sure we have capacity info else warn

Reply:
I don't think this is correct as (1) it will short-circuit to the `amount`-based check, and (2) if there is no capacity staked the entire returned object will be wrapped in `Option::None`, which gets translated by the `capacityInfo()` function using `unwrap_or()` to set `remainingCapacity` to 0.

Reply:
Yes, i see the object, however i am checking for `> 0` otherwise not
`[Nest] 35121  - 08/07/2023, 8:39:35 AM     LOG [QueueConsumerService] Capacity info: {"providerId":"1","currentBlockNumber":18,"nextEpochStart":100,"remainingCapacity":"0","totalCapacityIssued":"0"}`

Reply:
if its zero or undefined we should not go that route